docs/user/advanced.rst

```diff
@@ -102,6 +97,39 @@ You can then use the data in ``templates/index.html`` as you'd expect.
     </ul>
     {% endblock %}
 
+If data needs some work to produce then one can replace the dictionary
+by a function returning a dictionary. This function must either take no
+argument or one argument which will then be the current template.
+For instance the following gets the modification time of the template
+file into the context.
```

Comment:
As a matter of voice, I think we're trying to stick with the second person. I think this would read better as:

> For more complex data, you can use a function that returns a dictionary. This function may either take no arguments or  the current template as its sole argument. For example, the following add the modification time of the template file to the context.